packages/cubejs-api-gateway/src/helpers/shouldAddLimit.ts

```diff
@@ -0,0 +1,15 @@
+// Determine SQL query type and add LIMIT clause if needed
+export const shouldAddLimit = (sql: string): Boolean | null => {
+  // TODO: Enhance the way we determine query type
+  const ddlRegex = /^(CREATE|ALTER|DROP|TRUNCATE)\b/i;
+  const dmlRegex = /^(INSERT|UPDATE|DELETE)\b/i;
+  const selectRegex = /^(SELECT|WITH)\b/i;
+
+  if ((ddlRegex.test(sql)) || (dmlRegex.test(sql))) {
+    return false;
+  } else if (selectRegex.test(sql)) {
+    return true;
+  }
+  
+  return null;
```

Comment:
consider adding `throw new UserError('The query limit has been exceeded');` here instead of returning null